pkgs/applications/office/tradingview/default.nix

```
@@ -0,0 +1,65 @@
+{ fetchurl
+, lib
+, stdenv
+, squashfsTools
+, makeWrapper
+, electron
+, libsecret
+}:
+
+stdenv.mkDerivation rec {
+  pname = "tradingview";
+  version = "1.0.13";
+  revision = "24";
+
+  src = fetchurl {
+    url = "https://api.snapcraft.io/api/v1/snaps/download/nJdITJ6ZJxdvfu8Ch7n5kH5P99ClzBYV_${revision}.snap";
+    hash = "sha512-UZ8ogKdlmYkeIYC+skqxRPDd1KZYmPv+pbCW7XvvHsCs+ptkbOxaP+M82uHugmPf4aTUVU+0cI3hvjV2kTYkjA==";
+  };
+
+  nativeBuildInputs = [ squashfsTools makeWrapper ];
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  # Prevent double wrapping
+  dontWrapGApps = true;
+
+  unpackPhase = ''
+    runHook preUnpack
+    unsquashfs "$src" "resources/app.asar" "resources/app.asar.unpacked" "meta/gui"
+    cd squashfs-root
+    runHook postUnpack
+  '';
+
+  installPhase =
+    ''
+      runHook preInstall
+
```

```suggestion
  installPhase = ''
    runHook preInstall

```
please adjust formatting